pkg/driver/controller.go

```diff
@@ -235,13 +223,25 @@ func CreateZFSVolume(req *csi.CreateVolumeRequest) (string, error) {
 		return "", status.Error(codes.Internal, err.Error())
 	}
 
-	err = zfs.ProvisionVolume(volObj)
+	klog.Infof("zfs: trying volume creation %s/%s on node %s", pool, volName, prfList)
+
+	// try volume creation sequentially on all nodes
+	for _, node := range prfList {
+		vol, _ := volbuilder.BuildFrom(volObj).WithOwnerNode(node).WithVolumeStatus(zfs.ZFSStatusPending).Build()
+
+		err = zfs.ProvisionVolume(vol)
```

Comment:
@pawanpraka1 - overall the code seems to be good. 

In addition to the corner case mentioned in the PR (possibility of stale volumes), how about:
- cases, where the list returned, can get outdated because of parallel volume creation/deletion 
- validating the node itself is still intact while going through the list. How long do we wait before we determine that the current node selected is not suitable. 

I am now wondering if we should add an option where the provisioner can ask only for one node at a time and if the provisioning fails, return back so that CSI retries?

Reply:
> cases, where the list returned, can get outdated because of parallel volume creation/deletion

That should not be a problem here as the goal is to avoid getting stuck on a single node. So, even in this case scheduling might not be perfect but volume will get provision.

> validating the node itself is still intact while going through the list. How long do we wait before we determine that the current node selected is not suitable.

Yes, I think we should add wait here. Will add that.

> I am now wondering if we should add an option where the provisioner can ask only for one node at a time and if the provisioning fails, return back so that CSI retries?

That option is already there and controlled by allowedtopologies, it will try on the nodes which satisfy the topology contraints.